.github/workflows/staging-deployment.yaml

```diff
@@ -0,0 +1,37 @@
+name: staging-deployment
+# Trigger deployment only on push to develop branch
+on:
+  push:
+    branches:
+      - develop
+jobs:
+  deploy:
+    name: Deploy latest develop branch to staging
+    runs-on: ubuntu-latest
+    environment: staging
+    steps:
+      - name: Executing remote ssh commands using ssh key
+        uses: appleboy/ssh-action@v0.1.6
+        env:
+          DOCKER_TAG: develop
+          GITHUB_SHA: ${{ github.sha }}
+        with:
+          host: ${{ secrets.HOST_DNS }}
+          username: ${{ secrets.USERNAME }}
+          key: ${{ secrets.EC2_SSH_KEY }}
+          envs: DOCKER_TAG,GITHUB_SHA
+          command_timeout: 60m
+          script: |
+            echo "DOCKER_TAG: ${DOCKER_TAG}"
+            echo "GITHUB_SHA: ${GITHUB_SHA}"
+            docker system prune --force
+            cd ~/signoz
+            git status
+            git add .
+            git stash push -m "stashed on $(date --iso-8601=seconds)"
```

Comment:
@prashant-shahi what does this do?

Reply:
In case someone had previously added any uncommitted changes to the repository in the instance manually, we make sure to stash them with message including the ISO date.